src/app/clients/clients-view/view-survey/take-survey/take-survey.component.ts

```diff
@@ -0,0 +1,115 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+
+/** Custom Services */
+import { ClientsService } from '../../../clients.service';
+
+@Component({
+  selector: 'mifosx-take-survey',
+  templateUrl: './take-survey.component.html',
+  styleUrls: ['./take-survey.component.scss']
+})
+export class TakeSurveyComponent implements OnInit {
+
+  allSurveyData: any;
+  surveyData: any;
+  componentGroups: any[];
+  formData: {
+    userId: Number,
+    clientId: Number,
+    surveyId: Number,
+    scorecardValues: { questionId: Number, responseId: Number, value: String }[],
+    surveyName: String,
+    username: String,
+    id: Number
+  };
+
+  /**
+   * Retrieves the survey data from `resolve`.
+   * @param {ActivatedRoute} route Activated Route.
+   */
+  constructor(private route: ActivatedRoute,
+    private clientsService: ClientsService) {
+    this.route.data.subscribe((data: { allSurveysData: any }) => {
+      this.allSurveyData = data.allSurveysData;
+    });
+  }
+
+  onSurveyChange(re: any) {
```

Comment:
what is re?

Reply:
It is a variable receiving the event on change and displaying the questions as per the chosen survey.
I will be renaming all the variables where needed so that it conveys the use of it. No need to worry about that.